lib/curl_setup.h

```diff
@@ -471,6 +471,10 @@
 
 #  define DIR_CHAR      "\\"
 
+   /* Random generator shared between the Schannel vtls and Curl_rand*()
+      functions */
+   CURLcode curl_win32_random(unsigned char *entropy, size_t length);
```

Comment:
```suggestion
   CURLcode Curl_win32_random(unsigned char *entropy, size_t length);
```
and move to rand.h then include rand.h in schannel.c

Reply:
@jay: As for `crypt32`. This lib and `CryptGenRandom()` was available way back to Win95, and also WinCE. With one gotcha: the Unicode version `CryptAcquireContextW()` was not present in Win95, only `CryptAcquireContextA()` was. So if Win95-compatibility is important, we can force using the "ANSI" version, unconditionally even, as none of the exchanged values are Unicode strings:
```c
  #if !defined(UNDER_CE) && !defined(__CEGCC__) && !defined(__MINGW32CE__)  /* if WinCE-compatibility is a goal */
    #undef CryptAcquireContext
    #define CryptAcquireContext  CryptAcquireContextA
  #endif
```

To make MSVC support simpler and make it automatically apply to all external MSVC build environments, pulling in `crypt32` can be solved by adding this to `rand.c`:
```c
#ifdef WIN32
#include <wincrypt.h>
#ifdef _MSC_VER
#  pragma comment(lib, "crypt32.lib")
#endif
#endif
```
What do you think?

Reply:
> As for copying the declaration from a common header to each place of use. What is the reason we do this?

That should only be done if we for some reason can't just include the main header file with with the declaration.

Reply:
It should be documented that we use Curl_ for internally shared functions and curl_ publicly but I can't find it. @bagder?

Generally speaking each unit should have its own header (eg rand.c has rand.h) and that header should contain its Curl_ functions. (There are some exceptions to this but _generally_ it's true). If we put everything in a big header it would be messy, harder to manage and slow things down to include everything when you don't need everything. This way only units that need Curl_ functions from rand.c include rand.h.

Reply:
I mistakenly removed that section in a previous cleanup, coming back in #9037